test/extraction_test.jl

```diff
@@ -0,0 +1,39 @@
+@testset "Noise Extraction Test" begin
+  using Random, DiffEqNoiseProcess, DiffEqBase, Test
+  seed = 100
+  Random.seed!(seed)
+
+  tstart = 0.0
+  tend = 1.0
+  dt = 0.1
+  trange = (tstart, tend)
+  t = tstart:dt:tend
```

Comment:
ranges correct for this error, so instead of doing this, manually add

Reply:
Do you mean that the t range doesn't have the floating point error from  SciML/StochasticDiffEq.jl#324? I thought this is even necessary for the check. The time series in sol.t hosts the slightly off floating point numbers with the errors, so I am using the  t range  only for extracting the noise values at those times, and therefore I thought they should have this form.

Reply:
Ranges are specifically handled to do `t = tstart + i*dt` to avoid extra accumulation, so this isn't the best test

Reply:
yeah indeed,  a good point.. I decreased dt and checked if there are any issues.  For dt=0.001 (as on the current commit) all tests are passing. When using dt=0.0001 the solution array actually has size 10002, so the length checks will fail then. [ However, the important more one @test _sol == sol will still pass. ]

I also don't find any singularity in the t values for dt=0.0001:
findmin(sol.t[2:end]-sol.t[1:end-1]) # (9.999999999998899e-5, 1251)
findmax(sol.t[2:end]-sol.t[1:end-1]) # (0.00010000000000010001, 10001)

Reply:
>When using dt=0.0001 the solution array actually has size 10002, so the length checks will fail then

That's odd. Any indication as to why?

Reply:
probably similar floating point handling at the end of the integration

Reply:
Yes.. I also have the endpoint from https://github.com/SciML/StochasticDiffEq.jl/blob/master/src/integrators/integrator_utils.jl#L198-L214 under suspicion, I'll check later if I can fix it. Here is the associated dt plot: 
![plot](https://user-images.githubusercontent.com/42201748/86510373-5f2f7300-bdef-11ea-8c08-e33e38ddff58.png)
(because those are just floating point differences plot prints: `┌ Warning: No strict ticks found
└ @ PlotUtils ~/.julia/packages/PlotUtils/nCtbM/src/ticks.jl:283`)